x-pack/test/security_solution_api_integration/test_suites/entity_analytics/entity_store/trial_license_complete_tier/host_transform.ts

```diff
@@ -144,8 +128,58 @@ export default function (providerContext: FtrProviderContext) {
             expect(total.value).to.eql(1);
             const hit = result.hits.hits[0] as SearchHit<Ecs>;
             expect(hit._source).ok();
-            expect(hit._source?.host?.name).to.eql(HOST_NAME);
-            expect(hit._source?.host?.ip).to.eql(IPs);
+            expect(hit._source?.host?.name).to.eql(testDocs.name);
+            expect(hit._source?.host?.ip).to.eql(testDocs.ip);
+
+            return true;
+          }
+        );
+      });
+
+      it('Should successfully collect all expected fields', async () => {
+        const HOST_NAME: string = 'host-transform-test-all-fields';
+        const testDocs: HostTransformTestDocuments = {
+          name: HOST_NAME,
+          domain: ['example.com', 'sub.example.com'],
+          hostname: ['example.com', 'example.com'],
+          id: ['alpha', 'beta'],
+          osName: ['ubuntu', 'macos'],
+          osType: ['linux', 'darwin'],
+          mac: ['abc', 'def'],
+          arch: ['x86-64', 'arm64'],
+          ip: ['1.1.1.1', '2.2.2.2'],
+        };
+
+        await createDocumentsAndTriggerTransform(providerContext, testDocs, 2);
+
+        await retry.waitForWithTimeout(
+          'Document to be processed and transformed',
+          TIMEOUT_MS,
+          async () => {
+            const result = await es.search({
+              index: INDEX_NAME,
+              query: {
+                term: {
+                  'host.name': HOST_NAME,
+                },
+              },
+            });
+            const total = result.hits.total as SearchTotalHits;
+            expect(total.value).to.eql(1);
+            expect(result.hits.hits[0]._source).ok();
+            const source = result.hits.hits[0]._source as HostTransformResult;
+            expect(source.host).ok();
+            const hit = source.host as HostTransformResultHost;
+
+            expect(hit.name).to.eql(HOST_NAME);
+            expectFieldToEqualValues(hit.domain, testDocs.domain);
+            expectFieldToEqualValues(hit.hostname, ['example.com']);
+            expectFieldToEqualValues(hit.id, testDocs.id);
+            expectFieldToEqualValues(hit.os?.name, testDocs.osName);
+            expectFieldToEqualValues(hit.os?.type, testDocs.osType);
+            expectFieldToEqualValues(hit.ip, testDocs.ip);
+            expectFieldToEqualValues(hit.mac, testDocs.mac);
+            expectFieldToEqualValues(hit.architecture, testDocs.arch);
```

Comment:
It seems (from [here](https://github.com/elastic/kibana/blob/main/x-pack/solutions/security/plugins/security_solution/server/lib/entity_analytics/entity_store/entity_definitions/entity_descriptions/host.ts)) that we should also test the ´host.type´ field that is different from `host.os.type`